easybuild/easyblocks/t/tensorflow.py

```diff
@@ -0,0 +1,264 @@
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing TensorFlow, implemented as an easyblock
+"""
+import glob
+import os
+import stat
+import tempfile
+
+import easybuild.tools.environment as env
+import easybuild.tools.toolchain as toolchain
+from easybuild.easyblocks.generic.pythonpackage import PythonPackage
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, apply_regex_substitutions, mkdir, resolve_path
+from easybuild.tools.filetools import which, write_file
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+
+
+INTEL_COMPILER_WRAPPER = """#!/bin/bash
+export INTEL_LICENSE_FILE='%(intel_license_file)s'
+%(compiler_path)s "$@"
+"""
+
+
+class EB_TensorFlow(PythonPackage):
+    """Support for building/installing TensorFlow."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            # see https://developer.nvidia.com/cuda-gpus
+            'cuda_compute_capabilities': [[], "List of CUDA compute capabilities to build with", CUSTOM],
+            'with_mkl_dnn': [True, "Make TensorFlow use Intel MKL-DNN", CUSTOM],
+        }
+        return PythonPackage.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Custom configuration procedure for TensorFlow."""
+
+        tmpdir = tempfile.mkdtemp(suffix='-bazel-configure')
+
+        # Bazel reset environment in which build is performed, so $INTEL_LICENSE_FILE gets unset
+        # so, we create a wrapper for icc to make sure location of license server is available...
+        # cfr. https://github.com/bazelbuild/bazel/issues/663
+        if self.toolchain.comp_family() == toolchain.INTELCOMP:
+            icc_wrapper_txt = INTEL_COMPILER_WRAPPER % {
+                'compiler_path': which('icc'),
+                'intel_license_file': os.getenv('INTEL_LICENSE_FILE', os.getenv('LM_LICENSE_FILE')),
+            }
+            icc_wrapper = os.path.join(tmpdir, 'bin', 'icc')
+            write_file(icc_wrapper, icc_wrapper_txt)
+            adjust_permissions(icc_wrapper, stat.S_IXUSR)
+            env.setvar('PATH', ':'.join([os.path.dirname(icc_wrapper), os.getenv('PATH')]))
+            self.log.info("Using wrapper script for 'icc': %s", which('icc'))
```

Comment:
The wrapper creation is not listed in the extended dry run. I think it should be there too.